components/support/utils/src/main/java/mozilla/components/support/utils/Browsers.kt

```diff
@@ -0,0 +1,282 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.content.pm.PackageManager
+import android.net.Uri
+import java.util.HashMap
+
+/**
+ * Helpful tools for dealing with other browsers on this device.
+ *
+ * ```
+ * // Collect information about all installed browsers:
+ * val browsers = Browsers.all(context)
+ *
+ * // Collect information about installed browsers (and apps) that can handle a specific URL:
+ * val browsers = Browsers.forUrl(context, url)`
+ * ```
+ */
+class Browsers private constructor(
+    context: Context,
+    uri: Uri
+) {
+    /**
+     * Enum of known browsers and their package names.
+     */
+    enum class KnownBrowser constructor(
+        val packageName: String
+    ) {
+        FIREFOX("org.mozilla.firefox"),
+
+        FIREFOX_BETA("org.mozilla.firefox_beta"),
+        FIREFOX_AURORA("org.mozilla.fennec_aurora"),
+        FIREFOX_NIGHTLY("org.mozilla.fennec"),
+        FIREFOX_FDROID("org.mozilla.fennec_fdroid"),
+
+        FIREFOX_LITE("org.mozilla.rocket"),
+
+        FENIX_PREVIEW("org.mozilla.fenix"),
+        FENIX_DEBUG("org.mozilla.fenix.debug"),
+
+        REFERENCE_BROWSER("org.mozilla.reference.browser"),
+        REFERENCE_BROWSER_DEBUG("org.mozilla.reference.browser.debug"),
+
+        CHROME("com.android.chrome"),
+        CHROME_BETA("com.chrome.beta"),
+        CHROME_DEV("com.chrome.dev"),
+        CHROME_CANARY("com.chrome.canary"),
+        CHROME_LOCAL_BUILD("com.google.android.apps.chrome"),
+        CHROMIUM_LOCAL_BUILD("org.chromium.chrome"),
+
+        OPERA("com.opera.browser"),
+        OPERA_BETA("com.opera.browser.beta"),
+        OPERA_MINI("com.opera.mini.native"),
+        OPERA_MINI_BETA("com.opera.mini.native.beta"),
+
+        UC_BROWSER("com.UCMobile.intl"),
+        UC_BROWSER_MINI("com.uc.browser.en"),
+
+        ANDROID_STOCK_BROWSER("com.android.browser"),
+
+        SAMSUNG_INTERNET("com.sec.android.app.sbrowser"),
+        DOLPHIN_BROWSER("mobi.mgeek.TunnyBrowser"),
+        BRAVE_BROWSER("com.brave.browser"),
+        LINK_BUBBLE("com.linkbubble.playstore"),
+        ADBLOCK_BROWSER("org.adblockplus.browser"),
+        CHROMER("arun.com.chromer"),
+        FLYNX("com.flynx"),
+        GHOSTERY_BROWSER("com.ghostery.android.ghostery"),
+        DUCKDUCKGO("com.duckduckgo.mobile.android")
+    }
```

Comment:
Microsoft Edge-ium?

Reply:
This is the applicationId   `com.microsoft.emmx`

Reply:
Added!